frontend/src/ProfilePage.js

```diff
@@ -9,41 +11,68 @@ const backendHostURL = process.env.REACT_APP_BACKEND_HOST_URL
 class ProfilePage extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      hikeCard: [],
+      data:[]
+    };
   }
 
   componentDidMount() {
     fetch(`${backendHostURL}/login/${localStorage.getItem("_id")}`)
       .then((res) => res.json())
       .then((data) => {
-        console.log("data", data);
-        this.setState({ ...data[0] });
+        this.setState({ data: data[0] });
+        this.getHikeCards()
+        console.log(this.state.hikeCard)
       });
   }
 
+  getHikeCards() 
+  {
+    for (let i =0; i < this.state.data.saved_trails.length; i++)
+    {
+      const savedTrail = this.state.data.saved_trails[i]
+    
+      fetch(`${backendHostURL}/hike/${savedTrail}`, {
+      }).then((res) => res.json())
+      .then((data) => {
+        const added = this.state.hikeCard.concat(data[0])
+        this.setState({hikeCard: added})
```

Comment:
Does this mean we are setting the state for `hikeCard` multiple times until the loop ends? Just curious.

Reply:
Yeah, incrementally adding on to state.hikeCard was the only way I could get the HikeCardList component to read the array properly.

Reply:
Yeah, that's interesting. I'm guessing you tried to just append to a temporary list and then just set the state of the hikeCard list to the temporary one but that didn't work, which would probably take as much time as your current implementation now that I think about it. This added feature is really awesome. 

Also, it might help to rename the state variable to hikeCards since it holds the list.

Reply:
Yeah you're right hikeCards would make more sense since it can hold multiple hikes! I'll probably address that in a future PR

Reply:
addressed in https://github.com/MaggieYang57/CSC308-TeamB/pull/129/commits/0d2c481d5c5212ca67bbb472c2d5e205088055f2